com.kms.cura.dal/src/com/kms/cura/dal/database/DoctorUserDatabaseHelper.java

```diff
@@ -213,4 +216,130 @@ public DoctorUserEntity searchDoctor(UserEntity entity) throws ClassNotFoundExce
 				UserColumn.ID.getColumnName());
 	}
 
+	public List<DoctorUserEntity> searchDoctorFunction(DoctorSearchEntity search)
+			throws SQLException, ClassNotFoundException {
+		PreparedStatement stmt = null;
+		ResultSet rs = null;
+		List<DoctorUserEntity> result = new ArrayList<DoctorUserEntity>();
+
+		try {
+			stmt = getSearchStatement(search);
+
+			rs = stmt.executeQuery();
+
+			while (rs.next()) {
+				result.add((DoctorUserEntity) getEntityFromResultSet(rs));
+			}
+			return result;
+		} finally {
+			if (stmt != null) {
+				stmt.close();
+			}
+		}
+	}
+
+	protected PreparedStatement getSearchStatement(DoctorSearchEntity search) throws SQLException {
+		DoctorUserEntity doctor = search.getDoctor();
+		PreparedStatement stmt = null;
+		int count = 1;
+		stmt = con.prepareStatement(getSearchQuery(search));
+
+		String name = doctor.getName();
+		if (name == null) {
+			name = "";
+		}
+		stmt.setString(count, "%" + name + "%");
+		count++;
+
+		if (doctor.getSpeciality() != null && doctor.getSpeciality().size() > 0
+				&& !doctor.getSpeciality().get(0).equals("")) {
+			for (SpecialityEntity specialty : doctor.getSpeciality()) {
+				stmt.setString(count, specialty.getName());
+				count++;
+			}
+		}
+		if (doctor.getFacility() != null) {
+			String city = doctor.getFacility().get(0).getCity();
```

Comment:
Because there will only be one facility getting passed into here at a time. No multiple location allow for the search